cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -657,10 +658,10 @@ public QueryHandle deleteNamespace(Id.Namespace namespace) throws ExploreExcepti
       SessionHandle sessionHandle = openSession(sessionConf);
 
       String database = getHiveDatabase(namespace.getId());
-      String statement = String.format("DROP DATABASE %s", database);
+      String statement = String.format("DROP DATABASE %s CASCADE", database);
```

Comment:
hm... this seems dangerous. What if there is table in the database that is not owned by cdap?